### ISSUE TITLE
func kubernetes deploy doesn't support binding expressions

### ISSUE THREAD

**Issue by hannesne** [Azure/azure-functions-core-tools#2264]
The binding expressions (i.e. `"queueName": "%input-queue-name%"`) is set as-is in the deployment file that sets up the scaled object. It doesn't seem like keda supports this, leading to hard-to-diagnose issues when it tries to scale out the pods.

Some suggestions:
1. Call this out in the documentation.
1. Update the documentation to point out how one can use the --dry-run option to output a deploymentfile, which can be ~~hacked~~ updated and deployed manually using kubectl apply to get around this kind of issue.
1. Detect this scenario in the cli and display a warning when trying to deploy.
1. Try to resolve the binding expression at deploy time.
1. Update keda to support binding expressions.

**Comment by apawast** [comment]
@ahmelsayed can you please take a look at this?

**Comment by mcollier** [comment]
Could be related to https://github.com/kedacore/keda/issues/753.

**Comment by ahmelsayed** [comment]
Yes, I think it's related to the issue @mcollier linked to. Azure Functions Runtime resolves that value from the environment, so KEDA will need to do the same. However, this will only work if the environment variable is defined through kubernetes (env on the pod or a from a configMap or Secret) but not if it's defined in the docker image itself.
